code/modules/reagents/chemistry/machinery/chem_heater.dm

```diff
@@ -366,7 +366,7 @@
 
 	//trying to absorb buffer from currently inserted beaker
 	if(volume < 0)
-		var/datum/reagent/buffer_reagent = reagents.has_reagent(buffer_type)
+		var/datum/reagent/buffer_reagent = beaker.reagents.has_reagent(buffer_type)
```

Comment:
Few lines down, the acceptable amount to transfer is meant to be calculated with the INTERNAL reagent's maximum volume. With this fix merged:
![image](https://github.com/tgstation/tgstation/assets/25628932/fd5d8b81-ddf9-4498-a3eb-69910b9905c8)

Reply:
That's what I'm doing 
https://github.com/tgstation/tgstation/blob/548cad95963dbdbd0dfe2d640b520d12e31cc703/code/modules/reagents/chemistry/machinery/chem_heater.dm#L374
I'm using reagents.maximum_volume that is the internal buffer.

Reply:
You're also using `buffer_reagent.volume`, which was previously the internal buffer, but is now the external buffer.
Perhaps I should have elaborated about the screenshot, instead of trying to absorb `100 - stored volume` units of buffer, its trying to absorb `100 - provided volume` of buffer, which lets you fill beyond the intended 100 unit cap, and also refuses to absorb ANY buffer if the BEAKER is full.

Reply:
The old code for reference, which reasonably checks both the beaker's contents and the internal contents
https://github.com/tgstation/tgstation/blob/4f76fad2746edd2feedcf066a939e57b4cd38aae/code/modules/reagents/chemistry/machinery/chem_heater.dm#L458-L469

Reply:
Ohhh I see now. That was a very subtle bug, good catch pretty easy to fix